pkg/stanza/fileconsumer/benchmark_test.go

```diff
@@ -188,9 +188,12 @@ func BenchmarkFileInput(b *testing.B) {
 			cfg.PollInterval = time.Microsecond
 
 			doneChan := make(chan bool, len(files))
-			callback := func(_ context.Context, token emit.Token) error {
-				if len(token.Body) == 0 {
-					doneChan <- true
+			callback := func(_ context.Context, tokens []emit.Token) error {
+				for _, token := range tokens {
+					if len(token.Body) == 0 {
+						doneChan <- true
+						break
+					}
```

Comment:
nit: The empty body is just a hack to indicate EOF, but assuming ordering of lines is preserved, we could probably just check the last token in the slice. This might give us a better idea of the performance benefits.

Reply:
Thanks! Changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36663/commits/07e046515f65b2f7ee8a46d50ea3cf2096242342.